File: src/Implementations/CSharp/GameImpl/Bot.cs
using BotRace.Game.Runtime;

namespace BotRace.Game.Implementation
{
    public class Bot : BotRace.Game.Bot
    {
        private string uri;

        public Bot(string uri)
        {
            this.uri = uri;
        }

        public Movement Play()
        {
            return new Movement(Direction.W);
        }

        public void InvalidMoveResponse()
        {
            throw new System.NotImplementedException();
        }
    }
}

File: src/GameImpl/Bot.fs
namespace BotRace.Game.Implementation.FSharp
open BotRace.Game

type Bot(uri : string) =
    let u = uri

    interface BotRace.Game.Bot with
        member this.GetMoveRequest() = Direction.W |> Movement
        member this.SetCell cell = raise (new System.NotImplementedException())
        member this.InvalidMoveResponse() = raise (new System.NotImplementedException())

